client-mr/src/main/java/org/apache/hadoop/mapreduce/v2/app/RssMRAppMaster.java

```diff
@@ -239,6 +239,7 @@ public Thread newThread(Runnable r) {
       long heartbeatInterval = conf.getLong(RssMRConfig.RSS_HEARTBEAT_INTERVAL,
           RssMRConfig.RSS_HEARTBEAT_INTERVAL_DEFAULT_VALUE);
       long heartbeatTimeout = conf.getLong(RssMRConfig.RSS_HEARTBEAT_TIMEOUT, heartbeatInterval / 2);
+      client.registerApplicationInfo(appId, heartbeatTimeout, "user");
```

Comment:
Could we remove this if mr don't need to register application info?

Reply:
We need this because `applicationManager` remove expired app need `user`. When our Spark does not use the AppQuota checker, it also needs register application info, so it will not have much impact here.

Reply:
It seems that this isn't compatible feature. If we use old client to connect new service, something wrong will happen.

Reply:
Yes, because the old client does not have user information or registerApplicationInfo, it can only be updated at refreshApp, which I am compatible with.